docs/README.rst

```diff
@@ -19,45 +19,51 @@ The |kctlr-long| manages F5 BIG-IP `Local Traffic Manager <https://f5.com/produc
 Features
 --------
 - Dynamically creates, manages, and destroys BIG-IP objects.
-- Forwards traffic from BIG-IP to `Kubernetes clusters`_ via `NodePorts`_ or `ClusterIPs`_.
+- Forwards traffic from the BIG-IP device to `Kubernetes clusters`_ via `NodePorts`_ or `ClusterIPs`_.
 - Support for F5 `iApps`_.
 - Handles F5-specific VirtualServer objects created in Kubernetes.
-- Handles standard Ingress objects created in Kubernetes (with some F5-specific extensions).
+- Handles standard Kubernetes Ingress objects using F5-specific extensions.
+- Handles route configuration on the BIG-IP system (**OpenShift only**)
 
 Guides
 ------
-
-See the `F5 Container Connector for Kubernetes user documentation </containers/v1/kubernetes/>`_.
+See the |kctlr-long| `user documentation </containers/latest/kubernetes/>`_.
 
 Overview
 --------
-
 The |kctlr-long| is a Docker container that runs in a `Kubernetes`_ Pod.
-It uses an F5 Resource to determine:
+It uses `F5 Resource`_ s to determine:
 
-- what objects to configure on your BIG-IP, and
-- to which `Kubernetes Service`_ the BIG-IP objects belong.
+- what objects to configure on your BIG-IP system, and
+- to which `Kubernetes Service`_ those objects belong.
 
-The |kctlr-long| watches the Kubernetes API for the creation, modification or deletion of Kubernetes objects.  For some objects, it responds to these events by creating, modifying or deleting objects in the configuration of a BIG-IP. It handles these objects:
-- A *ConfigMap that is the F5-specific VirtualServer* type, used to create per-service virtual servers and/or pools on BIG-IP.
-- The standard Kubernetes *Ingress* object, used to create a single virtual server on BIG-IP with L7 policies to route to individual services. This object can have some F5-specific annotations documented below to control F5-specific behavior.
+The |kctlr| watches the Kubernetes API for the creation, modification, or deletion of Kubernetes objects.
+For some Kubernetes objects, the Controller responds by creating, modifying, or deleting objects in the BIG-IP system.
+The |kctlr| handles the following Kubernetes objects:
 
-One controller can handle a mix of these objects simultaneously. See below for the specifics regarding the handling of these objects.
+- A :ref:`virtual server F5 Resource` ConfigMap -- creates Service-specific front-end virtual servers and/or pools on the BIG-IP system.
+- The standard `Kubernetes Ingress`_ object -- creates a single front-end virtual server on the BIG-IP system that uses L7 policies to route to individual Services.
+  This object can have some F5-specific annotations (:ref:`documented below <ingress annotations>`) to control behavior specific to the |kctlr|.
+- An :ref:`iApp F5 Resource` ConfigMap -- deploys an iApp on the BIG-IP system.
 
```

Comment:
```
- The OpenShift Route object - create a single front-end virtual server on the BIG-IP system that uses L7 policies to route to individual Services.
  + This object can have some F5-specific annotations...
```

Maybe we want to remove some of these details and leave them in the specific sections for the particular resources.

What I'm asking for is basically a list of all of the things that k8s-bigip-ctlr watches to configure objects on BIG-IP (and then link out to the sections for each of these specific things):

- ConfigMaps that are F5's special VirtualServer type for basic load balancing
- ConfigMaps that are F5's special VirtualServer type for iApps
- Ingresses
- Routes (in OpenShift)

Reply:
Further complicating this a bit, there isn't a special "iApp F5 Resource" ConfigMap, at least not really.

What the docs are calling the "virtual server F5 Resource ConfigMap" and the "iapp F5 Resource ConfigMap" are the same objects in Kubernetes: ConfigMaps with the special f5type: virtual-server.

They're different only in the way you fill out the "frontend" section, not in the Kubernetes type.

It is basically two different frontend modes with little advantage to mixing/matching all the options, so I'm glad they have separate sections in the documentation.  I'm proposing describing them as "for basic load balancing" and "for iApps" instead of separate resource types.

Reply:
Addressed by omission, for the time being. We may want to circle back and expand on this later.